rust/src/transformations/manipulation/ffi.rs

```diff
@@ -1,95 +1,23 @@
-}
-
 #[no_mangle]
 pub extern "C" fn opendp_transformations__make_identity(
-    D: *const c_char,
-    M: *const c_char,
+    domain: *const AnyDomain,
+    metric: *const AnyMetric,
 ) -> FfiResult<*mut AnyTransformation> {
-    let M = try_!(Type::try_from(M));
-    let D = try_!(Type::try_from(D));
-
-    match &D.contents {
-        TypeContents::GENERIC { name, args } if name == &"VectorDomain" => {
-            if args.len() != 1 {
-                return err!(FFI, "VectorDomain only accepts one argument.").into();
-            }
-            let atomic_domain = try_!(Type::of_id(&args[0]));
-            let T = match atomic_domain.contents {
-                TypeContents::GENERIC { name, args } if name == "AtomDomain" => {
-                    if args.len() != 1 {
-                        return err!(FFI, "AtomDomain only accepts one argument.").into();
-                    }
-                    try_!(Type::of_id(&args[0]))
-                }
-                _ => return err!(FFI, "In FFI, make_identity's VectorDomain may only contain AtomDomain<_>").into()
-            };
-            fn monomorphize<M, T>() -> FfiResult<*mut AnyTransformation>
-                where M: 'static + Metric<Distance=IntDistance>,
-                      T: 'static + Clone + CheckAtom,
-                      (VectorDomain<AtomDomain<T>>, M): MetricSpace {
-                make_identity::<VectorDomain<AtomDomain<T>>, M>().into_any()
-            }
-            dispatch!(monomorphize, [
-                (M, [InsertDeleteDistance, SymmetricDistance]),
-                (T, @primitives)
-            ], ())
-        }
-        TypeContents::GENERIC { name, args } if name == &"AtomDomain" => {
-            if args.len() != 1 {
-                return err!(FFI, "AtomDomain only accepts one argument.").into();
-            }
-            let T = try_!(Type::of_id(&args[0]));
-
-            fn monomorphize<T>(M: Type) -> FfiResult<*mut AnyTransformation>
-                where T: 'static + DistanceConstant<T> + CheckAtom + One + Clone {
-                fn monomorphize<M>() -> FfiResult<*mut AnyTransformation>
-                    where M: 'static + Metric,
-                          M::Distance: CheckAtom + DistanceConstant<M::Distance> + One + Clone,
-                          (AtomDomain<M::Distance>, M): MetricSpace {
-                    make_identity::<AtomDomain<M::Distance>, M>().into_any()
-                }
-                dispatch!(monomorphize, [
-                    (M, [AbsoluteDistance<T>])
-                ], ())
-            }
-            dispatch!(monomorphize, [
-                (T, @numbers)
-            ], (M))
-        }
-        _ => err!(FFI, "Monomorphizations for the identity function are only available for VectorDomain<AtomDomain<_>> and AtomDomain<_>").into()
-    }
+    let domain = try_as_ref!(domain).clone();
+    let metric = try_as_ref!(metric).clone();
+    super::make_identity(domain, metric).into()
```

Comment:
Wow, good to get rid of all that! The issue with `MetricSpace` for `(AnyDomain, AnyMetric)` is tricky, not worth solving here. We don't really use `make_identity` anywhere except the sanity test, so I think just tagging it with a warning and `honest-but-curious` is a reasonable mitigation.